docs/source/software/systemwide/matlab.md

```diff
@@ -61,95 +61,372 @@ Use the following procedure if you need to install a MATLAB toolbox:
+
+```{code-block} matlab
+>> % Specify the partition 
+>> c.AdditionalProperties.Partition = 'partition-name';
+```
+
+Setting the number of cores for the job:
+
+```{code-block} matlab
+>> % Specify cores per node (default: 0)
+>> c.AdditionalProperties.ProcsPerNode = 4;
+```
+
+Setting exclusive use of the node:
+
+```{code-block} matlab
+>> % Set node exclusivity (default: false)
+>> c.AdditionalProperties.RequireExclusiveNode = true;
+```
 
-There are benchmarking scripts and examples available in **`**/shared/centos7/matlab/R2020a/examples/parallel/main/**`** on the cluster.
+```{note}
+The exclusive node option can increase your wait time in the queue so only user if required.
+```
 
-To make the scripts and examples available, add the path to this folder to the list of available paths by doing one of the following:
+If you are running the job on a reservation that has be setup:
 
-- On the MATLAB Home tab, in the **Environment** section, click **Set Path** and add the path to the script.
-- Alternatively, provide the script's full path in the MATLAB command line.
+```{code-block} matlab
+>> % Use reservation 
+>> c.AdditionalProperties.Reservation = 'reservation-name';
+```
 
-The contents of _myParallelAlgorithmFcn_ are as follows:
+Setting the time for the job to run on the HPC cluster:
 
 ```{code-block} matlab
-function [numWorkers,time] = myParallelAlgorithmFcn ()
+>> % Specify the wall time (e.g., 1 day, 5 hours, 30 minutes)
+>> c.AdditionalProperties.WallTime = '1-05:30';
+```
 
-complexities =  [2^18 2^20 2^21 2^22];
-numWorkers = [1 2 4 6 16 32 64];
+```{note}
+You need to make sure the time matches the partition you are submitting to or the jobs will not run.
+```
 
-time = zeros(numel(numWorkers),numel(complexities));
+### Using the MATLAB Client Interactively on the HPC Cluster
+If you want to run an interactive pool job on the cluster we can use a `parpool` like above:
 
-% To obtain predictable sequences of composite numbers, fix the seed
-% of the random number generator.
-rng(0,'twister');
+```{code-block} matlab
+>> % Get a handle to the cluster
+>> c = parcluster;
 
-for c = 1:numel(complexities)
+>> % Open a pool of 64 workers on the cluster
+>> pool = c.parpool(64);
+```
 
-   primeNumbers = primes(complexities(c));
-   compositeNumbers =    primeNumbers.*primeNumbers(randperm(numel(primeNumbers)));
-   factors = zeros(numel(primeNumbers),2);
+Instead of the open instance of MATLAB running this code, this will be run across the HPC cluster. Run the following code:
 
-   for w = 1:numel(numWorkers)
-       tic;
-       parfor (idx = 1:numel(compositeNumbers), numWorkers(w))
-          factors(idx,:) = factor(compositeNumbers(idx));
-       end
-       time(w,c) = toc;
+```{code-block} matlab
+>> % Run a parfor over 1000 iterations
+>> parfor idx = 1:1000
+      a(idx) = rand;
    end
```

Comment:
For the above lines, I get errors but it does run using the discovery profile.....

>> c = parcluster;
Error using parcluster
The JobStorageLocation of a cluster must be an accessible folder on the current machine's filesystem. The
value supplied (/home/s.caplins/.matlab/generic_cluster_jobs/discovery) is not a valid folder.
 
>> pool = c.parpool(64);
Unrecognized field name "parpool".
 
>> parfor idx = 1:1000
      a(idx) = rand;
   end
Starting parallel pool (parpool) using the 'Discovery' profile ...
>>

Reply:
this and all other commands below works after running configCluster